worker/upgradedatabase/shim.go

```diff
@@ -43,14 +43,24 @@ type pool struct {
 
 // IsPrimary (Pool) returns true if the Mongo primary is
 // running on the machine with the input ID.
-func (p *pool) IsPrimary(machineID string) (bool, error) {
+func (p *pool) IsPrimary(nodeID string) (bool, error) {
 	st := p.SystemState()
 
-	machine, err := st.Machine(machineID)
+	// For IAAS models, controllers are machines.
+	// For CAAS models, until we support HA, there's
+	// only one mongo and it's the primary.
+	model, err := st.Model()
 	if err != nil {
 		return false, errors.Trace(err)
 	}
+	if model.Type() == state.ModelTypeCAAS {
+		return true, nil
```

Comment:
I would like at least a comment here to indicate we know this is wrong if we ever do have HA CAAS. Otherwise, we have to audit most of the code base.

Reply:
done